DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/DllConflictManagment/DllConflictManager.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Autodesk.Revit.UI;
+
+namespace Speckle.Connectors.Revit.Plugin.DllConflictManagment;
+
+public class DllConflictManager
+{
+  private readonly Dictionary<string, AssemblyConflictInfo> _assemblyConflicts = new();
+  private readonly DllConflictManagmentOptionsLoader _optionsLoader;
+
+  public DllConflictManager(DllConflictManagmentOptionsLoader optionsLoader)
+  {
+    _optionsLoader = optionsLoader;
+  }
+
+  public void DetectConflicts()
+  {
+    Dictionary<string, Assembly> loadedAssembliesDict = new();
+    foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+    {
+      // TODO : what to do about multiple versions of loaded dlls?
+      loadedAssembliesDict[assembly.GetName().Name] = assembly;
+    }
+    LoadAssemblyAndDependencies(typeof(DllConflictManager).Assembly, loadedAssembliesDict, new HashSet<string>());
+  }
+
+  private void LoadAssemblyAndDependencies(
+    Assembly assembly,
+    Dictionary<string, Assembly> loadedAssemblies,
+    HashSet<string> visitedAssemblies
+  )
+  {
+    if (visitedAssemblies.Contains(assembly.GetName().Name))
+    {
+      return;
+    }
+    visitedAssemblies.Add(assembly.GetName().Name);
+
+    foreach (var assemblyName in assembly.GetReferencedAssemblies())
+    {
+      if (visitedAssemblies.Contains(assemblyName.Name))
+      {
+        continue;
+      }
+
+      if (loadedAssemblies.TryGetValue(assemblyName.Name, out Assembly? loadedAssembly))
+      {
+        if (loadedAssembly.GetName().Version != assemblyName.Version)
+        {
+          _assemblyConflicts[assemblyName.Name] = new(assemblyName, loadedAssembly);
+          continue; // if we already know there is a conflict here, no need to continue iterating dependencies
+        }
+      }
+      else
+      {
+        loadedAssembly = GetLoadedAssembly(assemblyName);
+        if (loadedAssembly is not null)
+        {
+          loadedAssemblies[assemblyName.Name] = loadedAssembly;
+        }
+      }
+
+      if (loadedAssembly is not null)
+      {
+        LoadAssemblyAndDependencies(loadedAssembly, loadedAssemblies, visitedAssemblies);
+      }
+    }
+  }
+
+  private static Assembly? GetLoadedAssembly(AssemblyName assemblyName)
+  {
+    try
+    {
+      return Assembly.Load(assemblyName);
+    }
+    catch (FileNotFoundException)
+    {
+      // POC : add logging
```

Comment:
hmmm, so it occurs to me we're doing this here in DUI3-DX but I think the intention was to do this for existing DUI2 - maybe this wasn't clear mind. POC: comments (let's have that spacing consistent so w can always find these POC: comments)

Do we know why we would get these exceptions? And why is it okay to swallow them?

Reply:
ah okay, I will make a new branch then